src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -318,7 +335,11 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                 }
             }
 
-            return $reflectionClass->newInstanceArgs($params);
+            if ($constructor->isConstructor()) {
```

Comment:
if `$constructor` is null, this code wouldn't be reached because of the if statement on line 303.